### ISSUE TITLE
pluck() should narrow value type to avoid ImplicitToStringCast

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#483]
## Description

When using `Builder::pluck()` or `Collection::pluck()` with a known string column, the returned Collection values are typed as `mixed`, causing `ImplicitToStringCast` when the result is used in a context expecting `string` values.

## Example

```php
Select::make('Plan', 'plan')
    ->options(static fn(): Collection => SomeModel::query()->pluck('name'));
```

`pluck('name')` returns `Collection<int, mixed>`. When these values flow into a context expecting `string` (e.g., Nova's `Select::options()` typed as `iterable<string|int, string|array>`), Psalm flags `ImplicitToStringCast` because `mixed` could be a `Stringable` object.

## Root cause

`Builder::pluck()` and `Collection::pluck()` return `Collection<array-key, mixed>` — the value type is not narrowed based on the column being plucked.

## Expected behavior

`pluck()` on an Eloquent Builder should ideally return `Collection<array-key, mixed>` at minimum, but Psalm should not flag the result as `ImplicitToStringCast` when used in a string context, since `pluck()` returns scalar database values (strings, ints, etc.), not objects with `__toString()`.

## Suggested fix

Consider typing `Builder::pluck()` return values more narrowly in the plugin stubs, e.g.:

```php
/**
 * @return \Illuminate\Support\Collection<array-key, scalar|null>
 */
public function pluck(string $column, ?string $key = null);
```

This would prevent `ImplicitToStringCast` since `scalar` doesn't include `Stringable` objects.
